### PR TITLE
Added golangci configuration and travis fix

### PR DESCRIPTION
**Pull request by PacoDw**
## Description
 
Implemented golangci.yml configuration file that will allow us to have more Lints for the code which helps to have more clean code and introduce good programming practices. Also, Travis was fixed.

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the MongoDB CLA
- [x] I have read the Terraform contribution guidelines 
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirments
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

File: .golangci.yml
Comment (gssbzn) [inline .golangci.yml]: [nit] enable all is a bit risky, new version could add new linters that may fail, is safer to be explicit about enabled linters

File: .golangci.yml
Comment (gssbzn) [inline .golangci.yml]: it would be better to do a `nolint: rule` in the offending blocks thsi way we prevent new offending blocks of being added

**Review by gssbzn — COMMENTED** [review]
I have a couple of nits since I recently did this for the atlas client

File: .golangci.yml
Comment (marinsalinas) [reply .golangci.yml]: Maybe we can use a similar GolangCI configuration in order to have a quality`standard` between MongoDB's golang repos. Thoughts? @themantissa  @PacoDw @thetonymaster

Comment (PacoDw) [reply .golangci.yml]: we totally agree, we already discuss this on slack xD

**Comment by themantissa** [comment]
Just noting:  Holding off on review until comments are addressed - seems we have a bit to adjust here first.  Thanks!

File: .golangci.yml
Comment (PacoDw) [reply .golangci.yml]: done

File: GNUmakefile
Comment (marinsalinas) [inline GNUmakefile:33]: Nice catch!

File: mongodbatlas/data_source_mongodbatlas_project.go
Comment (marinsalinas) [inline mongodbatlas/data_source_mongodbatlas_project.go]: found a typo.

File: mongodbatlas/resource_mongodbatlas_database_user.go
Comment (marinsalinas) [inline mongodbatlas/resource_mongodbatlas_database_user.go]: typo.

File: mongodbatlas/resource_mongodbatlas_global_cluster_config.go
Comment (marinsalinas) [inline mongodbatlas/resource_mongodbatlas_global_cluster_config.go]: typo.

File: mongodbatlas/resource_mongodbatlas_global_cluster_config.go
Comment (marinsalinas) [inline mongodbatlas/resource_mongodbatlas_global_cluster_config.go]: typo.

File: mongodbatlas/resource_mongodbatlas_global_cluster_config.go
Comment (marinsalinas) [inline mongodbatlas/resource_mongodbatlas_global_cluster_config.go]: typo

**Review by marinsalinas — CHANGES_REQUESTED** [review]
Just a few typos found, otherwise LGTM :)

File: GNUmakefile
Comment (PacoDw) [reply GNUmakefile:33]: Thanks :)

**Review by marinsalinas — APPROVED** [review]

**Review by themantissa — APPROVED** [review]
LGTM - please ensure we wait for @gssbzn to check his comments were resolved and approves before merging.

**Comment by PacoDw** [comment]
@gssbzn could you check if it's okay plz?

**Review by gssbzn — APPROVED** [review]
LGTM, thanks for doing all this

**Pull request merged**
